FirebaseAdmin/FirebaseAdmin.Tests/Auth/Jwt/MockTokenBuilder.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Apis.Util;
+using Xunit;
+
+namespace FirebaseAdmin.Auth.Jwt.Tests
+{
+    public sealed class MockTokenBuilder
+    {
+        internal string ProjectId { get; set; }
+
+        internal string TenantId { get; set; }
+
+        internal string Uid { get; set; }
+
+        internal string IssuerPrefix { get; set; }
+
+        internal IClock Clock { get; set; }
+
+        internal ISigner Signer { get; set; }
+
+        public async Task<string> CreateTokenAsync(
+            IDictionary<string, object> headerOverrides = null,
+            IDictionary<string, object> payloadOverrides = null)
```

Comment:
Just curious. Should the default of these collections be an empty collection instead of using null?
If they are empty we can avoid the null checks below.

Reply:
I just realized that the default values have to be constants at compile time... unless we use `headerOverrides = headerOverrides ?? new Dictionary <string, object>();`. Please ignore my first comment :)